src/core/maptoscreen.cpp

```diff
@@ -77,3 +78,36 @@ void MapToScreen::transformPoint()
   }
   emit screenPointChanged();
 }
+
+void MapToScreen::setMapDistance( const double distance )
+{
+  if ( mMapDistance == distance )
```

Comment:
Just thinking out loud: does it make sense to compare with `qgsDoubleNear()` here too? I guess not?

Reply:
I don't think so no, that's not a practice we've adopted elsewhere.